docs/reference/api/authentication.md

```diff
@@ -0,0 +1,130 @@
+
+Configure API keys as an option in the `@apostrophecms/express` module in `app.js`. Alternately, all `@apostrophecms/express` configuration may be added in a separate `modules/@apostrophecms/express/index.js` file.
+
+```javascript
+// In app.js
+require('apostrophe')({
+  modules: {
+    '@apostrophecms/express': {
+      options: {
+        apiKeys: {
+          // Use your own key value. Ideally use a strong, randomly generated
+          // key.
+          'myapikey1029384756': {
+            // The user role associated with this key
+            role: 'admin'
+          }
+        }
+      }
+    },
+    // ...
+```
+
+:::tip NOTE
+Right now `admin` is the only supported role value for API requests. Any other role will result in an anonymous request, which is the same as using no API key. This will change in the next alpha release when the rest of the permissions model comes in.
+:::
+
+### Usage
+
+Add an `authorization` [HTTP header](https://javascript.info/fetch#request-headers) to each request:
+
+```
+ApiKey myapikey1029384756
+```
+
+Alternatively you may pass the api key as the `apikey` or `apiKey` query parameter:
+
+```
+http://example.net/api/v1/article?apikey=myapikey1029384756
+```
+
+## Bearer tokens
+
+Bearer tokens are more appropriate for browser use and headless applications because they are tied to a single user account. To obtain a bearer token, make a POST request to `/api/v1/login/login`, with the `username` and `password` body properties:
+
+```javascript
+// Request inside an async function.
+const response = await fetch('http://example.net/api/v1/@apostrophecms/login/login', {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/x-www-form-urlencoded',
+  },
+  body: JSON.stringify({
+    username: 'your-username-here',
+    password: 'your-secure-password'
+  })
+});
+```
+
+### Response
+
+A successful response will return a JSON object with a `token` property.
+
+```javascript
+{
+  token: 'random123Token456xyz'
+}
+```
+
+### Usage
+
+Add this token to all REST API requests that require login by passing an `Authorization` header. This includes POST, PUT, PATCH and DELETE operations. You will also get more complete results from GET operations, including access to documents with `visibility` set to `loginRequired`.
+
+Your `authorization` HTTP header should look like:
+
+```
+Bearer random123Token456xyz
+```
+
+Note the need for the word "Bearer" before the key.
+
+To log out and destroy the token, send a POST request to `/api/v1/login/logout`, with the same `authorization` header. No body properties are required. After logging out, the token is no longer accepted.
+`
+
+## Session cookies
+
+You may also log in and use a session cookie, similarly to how Apostrophe itself maintains user sessions. To do so, include `session: true` in the POST request to `/api/v1/@apostrophecms/login/login`.
```

Comment:
It's not "similarly," it's literally how Apostrophe maintains user sessions. How about "As an alternative to a bearer token, you may request a session cookie. This is the mechanism Apostrophe's admin UI uses to log in. The advantage is that it persists automatically and works across tabs. The disadvantage is slightly more overhead on each request."

Reply:
+1, but I would amend the last big slightly "Session cookies will automatically persist across tabs, but there is slightly more overhead to each request". We might want to define "overhead" more clearly as well.

Reply:
On reflection, overhead isn't really the issue, it's about the same with bearer tokens; api keys are faster, because the server doesn't have to look anything up on each request that isn't already in memory, but it's not a difference that people should base their development decisions on.

The issue is more that session cookies are suitable for browser-based apps where the "same origin policy" feels right for your needs, and bearer tokens are more suitable to native apps and situations where you might want to intentionally use them across domains.